### PR TITLE
Making Date offset aware to align javascript date behavior on Pharo behavior

### PR DESCRIPTION
**Pull request by mattonem**
Basically adding an offset instVar in DateAndTime to store the off of the date. 
And then some hacky code to parse the timezone offset from the string and also print DateAndTime. 
Let me know what you think about it.

### CODE REVIEW

**Comment by mattonem** [comment]
related issue: https://github.com/PharoJS/PharoJS/issues/132

File: Pharo/PharoJs-Base-JS-CoreLibraries/PjDuration.class.st [hunk 1]
```diff
@@ -36,3 +36,20 @@ PjDuration >> addDateAndTime: aDateAndTime [
...
+
+	^ (self asNanoSeconds < 0
+		   ifTrue: [ '-' ]
+		   ifFalse: [ '+' ]) , (self printWithLeading0: self hoursCount)
+	  , ':' , (self printWithLeading0: self minutesCount)
+]
+
+{ #category : #adding }
+PjDuration >> printWithLeading0: aNumber [
+"there must be nicer solution for that..."
```
Comment (bouraqadi) [inline Pharo/PharoJs-Base-JS-CoreLibraries/PjDuration.class.st]:
```
^String stream contents: [: str |
	aNumber abs < 10
		  ifTrue: [str print: 0].
	str print: aNumber abs
]
```

File: Pharo/PharoJs-Base-JS-CoreLibraries/PjDuration.class.st [hunk 2]
```diff
@@ -36,3 +36,20 @@ PjDuration >> addDateAndTime: aDateAndTime [
 		setMilliseconds: (aDateAndTime getMilliseconds + self millisecondsCount);
 		yourself
 ]
+
+{ #category : #adding }
+PjDuration >> printString [
+
```
Comment (bouraqadi) [inline Pharo/PharoJs-Base-JS-CoreLibraries/PjDuration.class.st:42]: Use `String streamContents: [: str| ]`. It's cleaner and faster. See below.

File: Pharo/PharoJs-Base-JS-CoreLibraries/PjDateAndTime.class.st [hunk 3]
```diff
@@ -127,22 +165,23 @@ PjDateAndTime >> monthIndex: anInteger [
 	^self month: anInteger - 1
 ]
 
+{ #category : #accessing }
+PjDateAndTime >> offset [
```
Comment (bouraqadi) [inline Pharo/PharoJs-Base-JS-CoreLibraries/PjDateAndTime.class.st:169]: Provide a default value for `offset` as in
```st
PjDateAndTime >> offset
	^ offset ifNil: [self getTimezoneOffset]
```

File: Pharo/PharoJs-Base-JS-CoreLibraries/PjRxParser.class.st [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+}
+
+{ #category : #'pharoJs support' }
+PjRxParser class >> javascriptName [ 
+	^ 'RegExp'
+]
+
+{ #category : #testing }
+PjRxParser >> matchesIn: aString [
+	<javascript: 'return aString.match(this)'>
```
Comment (bouraqadi) [inline Pharo/PharoJs-Base-JS-CoreLibraries/PjRxParser.class.st]: Why not use Pharo `^ aString match: self`?

**Review by bouraqadi — CHANGES_REQUESTED** [review]
I like the idea of remembering the provided offset.

**Comment by mattonem** [comment]
Following your comments I made quite a few changes so we use `printOn:str` instead of `printStream` everywhere. 
Much better indeed.

File: Pharo/PharoJs-Base-JS-CoreLibraries/PjDateAndTime.class.st (hunk 3 above)
Comment (mattonem) [reply Pharo/PharoJs-Base-JS-CoreLibraries/PjDateAndTime.class.st:169]: For some reason, default offset is actual `-1 * self getTimezoneOffset`. 
I tried to understand why... but ended up just trusting the test cases...

File: Pharo/PharoJs-Base-JS-CoreLibraries/PjDuration.class.st (hunk 2 above)
Comment (mattonem) [reply Pharo/PharoJs-Base-JS-CoreLibraries/PjDuration.class.st:42]: Looks like Iceberg is not removing that method... Weird, i'll remove it from github interface

**Pull request merged**